Job[] jobs
Machine[] machines
int32 num_of_transporters